### DISCUSSION TITLE
New lines?

### DISCUSSION THREAD

**Discussion by JPR-source** [jspsych/jsPsych#916] [Q&A] [answered]
Hello,

In one of your tutorials you mention that javascript doesn't care about white space or line spacing.

So I'm wondering why this doesn't work (giving the error-Uncaught SyntaxError: '' string literal contains an unescaped line break):

stimulus: `'<div class="wrapper">`
                `<div class="box box1">-</div>`
                `<div class="box box2">-</div>`
                `<div class="box box3">-</div>`
                `<div class="box box4">-</div>`
`</div>'`

Whilst this does work:

stimulus: `'<div class="wrapper">'+`
                `'<div class="box box1">-</div>'+`
                `'<div class="box box2">-</div>'+`
               ` '<div class="box box3">-</div>'+`
                `'<div class="box box4">-</div>'+`
`'</div>'`

Are there some cases where you must indicate that you are going onto the next line and others in which you don't?

**Comment by vijaymarupudi** [comment, marked as answer]
JavaScript isn't entirely whitespace agnostic, @jodeleeuw probably meant that as *usually*.

An example is that semicolons are optional if you have a line break after a statement.

The other example is in strings. Sometimes you want a string with a line-break in it, or sometimes you want a break a string into multiple lines without a line break in it.

In order to not break legacy Javascript code, you can't just do what you wanted to do. Instead, you have to indicate to Javascript that this line-break doesn't mean anything, or you can add strings like the example you posted.

An example

```javascript
var x = "abc\
def\
ghi"

// x = "abcdefghi"
```

If you do want line-breaks in regular strings, you can use the `\n` sequence in them

```javascript
var x = "abc\ndef"
```

then x will be

```
abc
def
```

Obviously that's not as pretty. If you do want to include line-breaks in strings, and separate them into multiple lines, use string with a `` ` ``.

Example:

```javascript
var x = `abc
def`
```

then x will be

```
abc
def
```
